src/lwip/BUILD.gn

```diff
@@ -225,6 +225,12 @@ if (current_os == "zephyr" || current_os == "mbed") {
           [ "${chip_root}/third_party/lwip/repo/lwip/src/apps/mdns/mdns.c" ]
     } else if (lwip_platform == "silabs") {
       public_deps += [ "${efr32_sdk_build_root}:efr32_sdk" ]
+
+      sources += [
+        "${lwip_platform}/lwipopts-rs911x.h",
```

Comment:
Pretty sure theses files should be mutually exclusive..

Reply:
They should be however I was not sure on what flag. I will take a look.

Reply:
I added them. However I am unsure if correct: now we only layer-guard the files based on what compile options are chosen, however lwipopts.h relies on defines so it may include any of the files if the defines are odd and there is no way for us to force these.

Reply:
@jepenven-silabs 

Reverted that change: what happens is that lwipotps.h includes those files based on `#defines` and when gn checks includes, it does not know what is set or not. As a result, either I mark them as nogncheck (which seems to subvert things) or i add all files in the sources.

For now I would rather have all files in sources as this seems clearer.

Reply:
@andy31415 The header or its include has no reason to exist anymore. we can delete all this. I can do it in a separate PR if you want. Lwip is not used for our thread platforms

Reply:
I've just compared the files and there is really no reason to have a trio of configuration file. will accept your change but we should merge these three headers into a single one.